### ISSUE TITLE
Miner: Cryptonight@Nicehash doesn't accept any shares

### ISSUE THREAD

**Issue by antongi** [Bendr0id/xmrigCC#77]
Miner 1.50-1.52 shows no error, but zero shares good or total when using cryptonight nicehash pools. Windows 64-Bit gcc binaries. The old miner v1.4.0 works good.
![1 5 2](https://user-images.githubusercontent.com/16322954/37925953-f8f4ac2e-3103-11e8-860f-d1b5511f146e.png)
![1 5 2cc](https://user-images.githubusercontent.com/16322954/37925853-a9c37f90-3103-11e8-933b-c76ad22130a1.png)

**Comment by Bendr0id** [comment]
If nicehash wouldn't accept your shares the miner would print it out. 

The Difficulty is quite high, and the miner hashrate is 200 h/s which isnt that much. Couldn't it be that you just find valid shares within time?

**Comment by antongi** [comment]
But miner 1.4.0 works with the same config:
![1 4 0](https://user-images.githubusercontent.com/16322954/37927334-c77c1570-3107-11e8-83ec-7bf6b4da8bf0.png)

**Comment by Bendr0id** [comment]
Check this, not a single valid share within 10mins.

```
 * VERSIONS:     XMRigCC/1.4.0 (based on XMRig 2.4.4) libuv/1.8.0 gcc/7.2.0 (RELEASE)
 * HUGE PAGES:   available, enabled
 * CPU:          Unknown (1) x64 AES-NI
 * THREADS:      4, cryptonight, aes=1, hf=1, donate=5%
 * POOL #1:      cryptonight.eu.nicehash.com:3355
 * COMMANDS:     hashrate, pause, resume, quit
[2018-03-26 19:00:20] use pool cryptonight.eu.nicehash.com:3355 169.51.48.177
[2018-03-26 19:00:20] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:00:38] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:00:52] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:01:18] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:01:21] speed 2.5s/60s/15m 56.0 60.8 n/a H/s max: 66.1 H/s
[2018-03-26 19:01:24] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:02:21] speed 2.5s/60s/15m 75.7 70.2 n/a H/s max: 73.6 H/s
[2018-03-26 19:03:21] speed 2.5s/60s/15m 54.9 67.3 n/a H/s max: 73.8 H/s
[2018-03-26 19:04:21] speed 2.5s/60s/15m 71.0 67.3 n/a H/s max: 75.1 H/s
[2018-03-26 19:04:52] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:05:11] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:05:21] speed 2.5s/60s/15m 66.2 65.6 n/a H/s max: 75.1 H/s
[2018-03-26 19:06:21] speed 2.5s/60s/15m 54.1 66.1 n/a H/s max: 75.1 H/s
[2018-03-26 19:07:21] speed 2.5s/60s/15m 67.3 65.5 n/a H/s max: 75.1 H/s
[2018-03-26 19:07:21] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:07:44] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:07:59] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:08:21] speed 2.5s/60s/15m 64.9 64.6 n/a H/s max: 75.1 H/s
[2018-03-26 19:08:33] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:08:54] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:09:09] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:09:21] speed 2.5s/60s/15m 57.1 66.0 n/a H/s max: 75.1 H/s
[2018-03-26 19:09:37] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:10:21] speed 2.5s/60s/15m n/a 65.9 n/a H/s max: 75.1 H/s
[2018-03-26 19:10:24] use pool cryptonight.eu.nicehash.com:3355 169.51.48.177
[2018-03-26 19:10:24] new job from cryptonight.eu.nicehash.com:3355 diff 200007
[2018-03-26 19:10:31] new job from cryptonight.eu.nicehash.com:3355 diff 200007
```

**Comment by Bendr0id** [comment]
It's a pure luck-shoot. 

Make sure nicehash is set to true in the config on all configured pools.

If you want to force XMRigCC to behave excatly the same like 1.4.0 put `"force-pow-version" : 1` in your config.

**Comment by antongi** [comment]
I just did a test:
1.5.2 miner with `"force-pow-version" : 1` - no shares accepted after 30 minutes
1.4.0 - first share accepted after 10 minutes
2.5.2 new original XMRig with default `"variant": -1` - first share accepted after 2-3 minutes

Maybe it is a luck or Nicehash doesn't play _nice_ with CC miners now, but I guess I'll stick to 2.5.2 version for now, though I like the convenience of CC.

**Comment by sebastianstolzenberg** [comment]
At difficulty 200000 with a rate of 188 H/S you should hit a valid hash about every 18 minutes, in average. Since valid nonces are randomly distributed, deviations as you've seen them are totally normal.  At that difficulty you can only judge on the miner after it has run for a few hours. Just count the total number hashes of found after e.g. 12 hours. Actually I would expect xmrigCC to show a slightly higher hash rate than the competition.

But, with a 188 H/S miner, if possible consider getting lower difficulty jobs from the pool. Otherwise put a proxy between pool and miner. The proxy reduces the difficulty for the miner, and by that allows it to submit more often.

**Comment by antongi** [comment]
Thanks for the detailed explanation!
As far as I read you cannot use xmrig proxy with Nicehash.
I did a prolonged test: **XMRigCC/1.5.2 had 1 (one!) accepted and 3 rejected shares in 3 hours**. I didn't want to waste more CPU time, and so switched to **XMRigCC/1.4.0: 10 accepted and 0 rejected shares in 2 hours**.
Same rig, same config.

**Comment by sebastianstolzenberg** [comment]
Thank you for sacrificing your CPU time. The problem is obvious now and will be investigated ...

**Comment by antongi** [comment]
OK, I did a 2 hours run with **XMRigCC/1.5.0** - 0 shares accepted by nicehash pool, 2 rejected:
```
rejected (0/1) diff 200007 "Invalid nonce; is miner not compatible with NiceHash?" (539518 ms)
rejected (0/2) diff 200007 "Job not found." (1546858 ms)
```

**Comment by Bendr0id** [comment]
Thanks. I'm already on it and located the problem.. I hope to have a fixed soon.

Am 28.03.2018 19:10 schrieb antongi <notifications@github.com>:

OK, I did a 2 hours run with XMRigCC/1.5.0 - 0 shares accepted by nicehash pool, 2 rejected:

rejected (0/1) diff 200007 "Invalid nonce; is miner not compatible with NiceHash?" (539518 ms)
rejected (0/2) diff 200007 "Job not found." (1546858 ms)

**Comment by Bendr0id** [comment]
@antongi 

Could you please try to fetch that branch `pool_variant_and_fixed_cpu_autodetect` if it fixed the issue for you?

**Comment by antongi** [comment]
Please make Windows 64-Bit binaries, and I'll test it.

**Comment by Bendr0id** [comment]
There you go:

https://ci.appveyor.com/project/Bendr0id/xmrigcc/build/1.5.4/artifacts

**Comment by antongi** [comment]
This build stats for 1 hour: 4 shares accepted, 0 rejected, 4 errors: "end of file" -> close connection.

**Comment by Bendr0id** [comment]
The end of file error now are only visible. Before there wasn't a log.. so all is good. Are you happy with the shares? For me it now looks like same performance then with 1.4.0 right?

**Comment by antongi** [comment]
Yes, looks like the same performance as with 1.4.0. Thanks!

**Comment by Bendr0id** [comment]
Ok will create a release soon

Am 29.03.2018 19:33 schrieb antongi <notifications@github.com>:

Yes, looks like the same performance as with 1.4.0. Thanks!

**Comment by Bendr0id** [comment]
Merged to master **Release v1.5.3 (#81)**
